ui/pages/confirmations/components/confirm/info/typed-sign/typed-sign.tsx

```diff
@@ -27,9 +29,11 @@ const TypedSignInfo: React.FC = () => {
     return null;
   }
 
-  const { domain = {} } = JSON.parse(
-    currentConfirmation.msgParams.data as string,
-  );
+  const {
+    domain,
+    domain: { verifyingContract },
+    primaryType,
+  } = JSON.parse(currentConfirmation.msgParams.data as string);
```

Comment:
Can we use the `parseTypedDataMessage` util here to encapsulate the parsing of the message?

Reply:
That also involves sanitizing the message which is not needed here.

Reply:
Could we split the sanitise into a separate util?

Currently we have to separate calls to `JSON.parse` reading some of the same fields back, that feels like a good case for abstraction?

Reply:
That will be creating a function above `JSON.parse`.

Reply:
It would, but more importantly it alone would know what properties are in the typed data and how to get it out, meaning both usages could use it and just destructure the properties they need.

Reply:
👍 I updated the PR